### PR TITLE
UCT/IB/MLX5: Check device is mlx5

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8230/ for details (Mellanox internal link).

File: src/uct/ib/mlx5/ib_mlx5_dv.c [hunk 1]
```diff
@@ -33,6 +33,10 @@ static ucs_status_t uct_ib_mlx5_device_init(uct_ib_device_t *dev)
     struct ibv_cq *cq;
     struct ibv_qp *qp;
 
+    if (!(uct_ib_device_spec(dev)->flags & UCT_IB_DEVICE_FLAG_MLX5_PRM)) {
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5_dv.c]: @Artemy-Mellanox this means we would not be able to remove the device table in upstream driver, right? Is there something that need to fix in rdma-core/MLNX_OFED? If yes pls open internal RM issue for Majd

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5_dv.c]: AFAIU we should call mlx5dv* functions only if we know there is mlx5 provider behind. I still don't know what is the correct way to check this, so using what we have currently.

Comment (yosefe) [reply src/uct/ib/mlx5/ib_mlx5_dv.c]: i think that as general API, mlx5dv should be provide a query-if-supported functionality.
@jgunthorpe what do you think?

Comment (jgunthorpe) [reply src/uct/ib/mlx5/ib_mlx5_dv.c]: There is a gap in the DV definition, there is no way for the user to know if the DV works with the ibv_context, and calling DV functions with a mismatched context will crash. This is something we really should address..

**Pull request merged**